arch/src/x86_64/mod.rs

```diff
@@ -166,7 +166,7 @@ pub fn configure_system(
         add_e820_entry(
             &mut params.0,
             himem_start.raw_value(),
-            mem_end.unchecked_offset_from(himem_start),
+            mem_end.unchecked_offset_from(himem_start) + 1,
```

Comment:
Yes the last byte was missing, and you could say in the commit message that's because we're providing the range size here.